### ISSUE TITLE
Missing go dependencies mean internet access is required to compile BPM v1.1.11

### ISSUE THREAD

**Issue by peterellisjones** [cloudfoundry/bpm-release#149]
Hi Folks,

It looks like an unvendored dependency is causing the compilation step to fail for us using the latest release. I can see that go.sum added [github.com/checkpoint-restore/go-criu/v5](https://github.com/cloudfoundry/bpm-release/blob/55f7c028169825f61ec7c9fe40c91f0c8bb39ea4/src/bpm/go.sum#L11) six days ago, but this is not present in [vendor/github.com](https://github.com/cloudfoundry/bpm-release/tree/55f7c028169825f61ec7c9fe40c91f0c8bb39ea4/src/bpm/vendor/github.com)

This is causing the [go build](https://github.com/cloudfoundry/bpm-release/blob/master/packages/bpm-runc/packaging#L94-L99) packaging step to fail for us in environments that do not have internet access when it tries to download the missing dependencies -- apologies for the screenshot, I'm unable to copy and paste logs directly from this environment.

![Screenshot 2021-05-20 at 08 04 16](https://user-images.githubusercontent.com/2788556/118934207-02bf3e80-b942-11eb-94d9-b8aa8ca2a182.png)

cheers

Pete

**Comment by peterellisjones** [comment]
Hi folks, here's the bosh output for compile bpm-runc step when compiling BPM 1.1.11 on bionic stemcell v1.1

```
Task 51 | 00:22:06 | Error: Action Failed get_task: Task 7c3b6167-d143-4ba9-7a75-2a51a1b9fffb result: Compiling package bpm-runc: Running packaging script: Running packaging script: Command exited with 1; Truncated stdout: checking whether make supports nested variables... yes

checking how to create a pax tar archive... gnutar

checking for gcc... gcc

checking whether the C compiler works... yes

checking for C compiler default output file name... a.out

checking for suffix of executables... 

checking whether we are cross compiling... no

checking for suffix of object files... o

checking whether we are using the GNU C compiler... yes

checking whether gcc accepts -g... yes

checking for gcc option to accept ISO C89... none needed

checking whether gcc understands -c and -o together... yes

checking whether make supports the include directive... yes (GNU style)

checking dependency style of gcc... gcc3

checking for ar... ar

checking the archiver (ar) interface... ar

checking build system type... x86_64-pc-linux-gnu

checking host system type... x86_64-pc-linux-gnu

checking how to print strings... printf

checking for a sed that does not truncate output... /bin/sed

checking for grep that handles long lines and -e... /bin/grep

checking for egrep... /bin/grep -E

checking for fgrep... /bin/grep -F

checking for ld used by gcc... /usr/bin/ld

checking if the linker (/usr/bin/ld) is GNU ld... yes

checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B

checking the name lister (/usr/bin/nm -B) interface... BSD nm

checking whether ln -s works... yes

checking the maximum length of command line arguments... 1572864

checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop

checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop

checking for /usr/bin/ld option to reload object files... -r

checking for objdump... objdump

checking how to recognize dependent libraries... pass_all

checking for dlltool... no

checking how to associate runtime and link libraries... printf %s\n

checking for archiver @FILE support... @

checking for strip... strip

checking for ranlib... ranlib

checking command to parse /usr/bin/nm -B output from gcc object... ok

checking for sysroot... no

checking for a working dd... /bin/dd

checking how to truncate binary pipes... /bin/dd bs=4096 count=1

checking for mt... mt

checking if mt is a manifest tool... no

checking how to run the C preprocessor... gcc -E

checking for ANSI C header files... yes

checking for sys/types.h... yes

checking for sys/stat.h... yes

checking for stdlib.h... yes

checking for string.h... yes

checking for memory.h... yes

checking for strings.h... yes

checking for inttypes.h... yes

checking for stdint.h... yes

checking for unistd.h... yes

checking for dlfcn.h... yes

checking for objdir... .libs

checking if gcc supports -fno-rtti -fno-exceptions... no

checking for gcc option to produce PIC... -fPIC -DPIC

checking if gcc PIC flag -fPIC -DPIC works... yes

checking if gcc static flag -static works... yes

checking if gcc supports -c -o file.o... yes

checking if gcc supports -c -o file.o... (cached) yes

checking whether the gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes

checking whether -lc should be explicitly linked in... no

checking dynamic linker characteristics... GNU/Linux ld.so

checking how to hardcode library paths into programs... immediate

checking whether stripping libraries is possible... yes

checking if libtool supports shared libraries... yes

checking whether to build shared libraries... yes

checking whether to build static libraries... yes

checking whether make supports nested variables... (cached) yes

checking for linux/seccomp.h... yes

checking for cython3... no

checking for cython... no

checking for gperf... gperf

checking for cov-build... no

checking whether to build with code coverage support... no

checking that generated files are newer than configure... done

configure: creating ./config.status

config.status: creating libseccomp.pc

config.status: creating include/seccomp.h

config.status: creating Makefile

config.status: creating include/Makefile

config.status: creating src/Makefile

config.status: creating src/python/Makefile

config.status: creating tools/Makefile

config.status: creating tests/Makefile

config.status: creating doc/Makefile

config.status: creating configure.h

config.status: executing depfiles commands

config.status: executing libtool commands

make --quiet --no-print-directory all-recursive

Making all in include

Making all in src

Making all in .

  CC       libseccomp_la-api.lo

  CC       libseccomp_la-system.lo

  CC       libseccomp_la-helper.lo

  CC       libseccomp_la-gen_pfc.lo

  CC       libseccomp_la-gen_bpf.lo

  CC       libseccomp_la-hash.lo

  CC       libseccomp_la-db.lo

  CC       libseccomp_la-arch.lo

  CC       libseccomp_la-arch-x86.lo

  CC       libseccomp_la-arch-x86_64.lo

  CC       libseccomp_la-arch-x32.lo

  CC       libseccomp_la-arch-arm.lo

  CC       libseccomp_la-arch-aarch64.lo

  CC       libseccomp_la-arch-mips.lo

  CC       libseccomp_la-arch-mips64.lo

  CC       libseccomp_la-arch-mips64n32.lo

  CC       libseccomp_la-arch-parisc.lo

  CC       libseccomp_la-arch-parisc64.lo

  CC       libseccomp_la-arch-ppc.lo

  CC       libseccomp_la-arch-ppc64.lo

  CC       libseccomp_la-arch-riscv64.lo

  CC       libseccomp_la-arch-s390.lo

  CC       libseccomp_la-arch-s390x.lo

  CC       libseccomp_la-syscalls.lo

  CC       libseccomp_la-syscalls.perf.lo

  CCLD     libseccomp.la

Making all in tools

  CC       scmp_sys_resolver.o

  CCLD     scmp_sys_resolver

  CC       scmp_arch_detect.o

  CCLD     scmp_arch_detect

  CC       scmp_bpf_disasm.o

  CC       util.lo

  CCLD     util.la

  CCLD     scmp_bpf_disasm

  CC       scmp_bpf_sim.o

  CCLD     scmp_bpf_sim

  CC       scmp_api_level.o

  CCLD     scmp_api_level

Making all in tests

Making all in doc

Making install in include

 /bin/mkdir -p '/var/vcap/packages/bpm-runc/lib/libseccomp/include'

 /usr/bin/install -c -m 644 seccomp.h seccomp-syscalls.h '/var/vcap/packages/bpm-runc/lib/libseccomp/include'

Making install in src

Making install in .

 /bin/mkdir -p '/var/vcap/packages/bpm-runc/lib/libseccomp/lib'

 /bin/bash ../libtool   --mode=install /usr/bin/install -c   libseccomp.la '/var/vcap/packages/bpm-runc/lib/libseccomp/lib'

libtool: install: /usr/bin/install -c .libs/libseccomp.so.2.5.1 /var/vcap/packages/bpm-runc/lib/libseccomp/lib/libseccomp.so.2.5.1

libtool: install: (cd /var/vcap/packages/bpm-runc/lib/libseccomp/lib && { ln -s -f libseccomp.so.2.5.1 libseccomp.so.2 || { rm -f libseccomp.so.2 && ln -s libseccomp.so.2.5.1 libseccomp.so.2; }; })

libtool: install: (cd /var/vcap/packages/bpm-runc/lib/libseccomp/lib && { ln -s -f libseccomp.so.2.5.1 libseccomp.so || { rm -f libseccomp.so && ln -s libseccomp.so.2.5.1 libseccomp.so; }; })

libtool: install: /usr/bin/install -c .libs/libseccomp.lai /var/vcap/packages/bpm-runc/lib/libseccomp/lib/libseccomp.la

libtool: install: /usr/bin/install -c .libs/libseccomp.a /var/vcap/packages/bpm-runc/lib/libseccomp/lib/libseccomp.a

libtool: install: chmod 644 /var/vcap/packages/bpm-runc/lib/libseccomp/lib/libseccomp.a

libtool: install: ranlib /var/vcap/packages/bpm-runc/lib/libseccomp/lib/libseccomp.a

libtool: finish: PATH="/var/vcap/packages/bpm-runc/gperf/bin:/var/vcap/bosh/bin:/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin:/sbin" ldconfig -n /var/vcap/packages/bpm-runc/lib/libseccomp/lib

----------------------------------------------------------------------

Libraries have been installed in:

   /var/vcap/packages/bpm-runc/lib/libseccomp/lib


If you ever happen to want to link against installed libraries

in a given directory, LIBDIR, you must either use libtool, and

specify the full pathname of the library, or use the '-LLIBDIR'

flag during linking and do at least one of the following:

   - add LIBDIR to the 'LD_LIBRARY_PATH' environment variable

     during execution

   - add LIBDIR to the 'LD_RUN_PATH' environment variable

     during linking

   - use the '-Wl,-rpath -Wl,LIBDIR' linker flag

   - have your system administrator add LIBDIR to '/etc/ld.so.conf'


See any operating system documentation about shared libraries for

more information, such as the ld(1) and ld.so(8) manual pages.

----------------------------------------------------------------------

Making install in tools

 /bin/mkdir -p '/var/vcap/packages/bpm-runc/lib/libseccomp/bin'

  /bin/bash ../libtool   --mode=install /usr/bin/install -c scmp_sys_resolver '/var/vcap/packages/bpm-runc/lib/libseccomp/bin'

libtool: install: /usr/bin/install -c .libs/scmp_sys_resolver /var/vcap/packages/bpm-runc/lib/libseccomp/bin/scmp_sys_resolver

Making install in tests

Making install in doc

 /bin/mkdir -p '/var/vcap/packages/bpm-runc/lib/libseccomp/share/man/man1'

 /usr/bin/install -c -m 644 man/man1/scmp_sys_resolver.1 '/var/vcap/packages/bpm-runc/lib/libseccomp/share/man/man1'

 /bin/mkdir -p '/var/vcap/packages/bpm-runc/lib/libseccomp/share/man/man3'

 /usr/bin/install -c -m 644 man/man3/seccomp_arch_add.3 man/man3/seccomp_arch_exist.3 man/man3/seccomp_arch_native.3 man/man3/seccomp_arch_remove.3 man/man3/seccomp_arch_resolve_name.3 man/man3/seccomp_attr_get.3 man/man3/seccomp_attr_set.3 man/man3/seccomp_export_bpf.3 man/man3/seccomp_export_pfc.3 man/man3/seccomp_init.3 man/man3/seccomp_load.3 man/man3/seccomp_merge.3 man/man3/seccomp_release.3 man/man3/seccomp_reset.3 man/man3/seccomp_rule_add.3 man/man3/seccomp_rule_add_array.3 man/man3/seccomp_rule_add_exact.3 man/man3/seccomp_rule_add_exact_array.3 man/man3/seccomp_notify_alloc.3 man/man3/seccomp_notify_fd.3 man/man3/seccomp_notify_free.3 man/man3/seccomp_notify_id_valid.3 man/man3/seccomp_notify_receive.3 man/man3/seccomp_notify_respond.3 man/man3/seccomp_syscall_priority.3 man/man3/seccomp_syscall_resolve_name.3 man/man3/seccomp_syscall_resolve_name_arch.3 man/man3/seccomp_syscall_resolve_name_rewrite.3 man/man3/seccomp_syscall_resolve_num_arch.3 man/man3/seccomp_version.3 man/man3/seccomp_api_get.3 man/man3/seccomp_api_set.3 '/var/vcap/packages/bpm-runc/lib/libseccomp/share/man/man3'

 /bin/mkdir -p '/var/vcap/packages/bpm-runc/lib/libseccomp/lib/pkgconfig'

 /usr/bin/install -c -m 644 libseccomp.pc '/var/vcap/packages/bpm-runc/lib/libseccomp/lib/pkgconfig'

/var/vcap/packages/bpm-runc/src/github.com/opencontainers/runc /var/vcap/data/compile/bpm-runc

, Stderr: + RUNC_VERSION=1.0.0-rc95

+ PKG_CONFIG_VERSION=0.29.2

++ mktemp -d

+ PKG_CONFIG_PREFIX=/var/vcap/data/tmp/tmp.qVucM4AA3a

+ GPERF_VERSION=3.1

+ GPERF_PREFIX=/var/vcap/packages/bpm-runc/gperf

+ LIBSECCOMP_PREFIX=/var/vcap/packages/bpm-runc/lib/libseccomp

+ LIBSECCOMP_RUNTIME_PATH=/var/vcap/packages/bpm/lib/libseccomp/lib

+ LIBSECCOMP_VERSION=2.5.1

+ RUNC_PACKAGE_PATH=github.com/opencontainers/runc

+ set -eu

+ tar xf pkg-config/pkg-config-0.29.2.tar.gz

+ cd pkg-config-0.29.2

+ ./configure --prefix=/var/vcap/data/tmp/tmp.qVucM4AA3a --with-internal-glib

+ make

ar: `u' modifier ignored since `D' is the default (see `U')

gbacktrace.c: In function 'g_on_error_query':

gbacktrace.c:177:5: warning: ignoring return value of 'fgets', declared with attribute warn_unused_result [-Wunused-result]

     fgets (buf, 8, stdin);

     ^~~~~~~~~~~~~~~~~~~~~

gbacktrace.c: In function 'stack_trace':

gbacktrace.c:301:18: warning: ignoring return value of 'dup', declared with attribute warn_unused_result [-Wunused-result]

       close (0); dup (in_fd[0]);   /* set the stdin to the in pipe */

                  ^~~~~~~~~~~~~~

gbacktrace.c:302:18: warning: ignoring return value of 'dup', declared with attribute warn_unused_result [-Wunused-result]

       close (1); dup (out_fd[1]);  /* set the stdout to the out pipe */

                  ^~~~~~~~~~~~~~~

gbacktrace.c:303:18: warning: ignoring return value of 'dup', declared with attribute warn_unused_result [-Wunused-result]

       close (2); dup (out_fd[1]);  /* set the stderr to the out pipe */

                  ^~~~~~~~~~~~~~~

gbacktrace.c:318:3: warning: ignoring return value of 'write', declared with attribute warn_unused_result [-Wunused-result]

   write (in_fd[1], "backtrace\n", 10);

   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

gbacktrace.c:319:3: warning: ignoring return value of 'write', declared with attribute warn_unused_result [-Wunused-result]

   write (in_fd[1], "p x = 0\n", 8);

   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

gbacktrace.c:320:3: warning: ignoring return value of 'write', declared with attribute warn_unused_result [-Wunused-result]

   write (in_fd[1], "quit\n", 5);

   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~

In file included from ggettext.c:30:0:

ggettext.c: In function 'ensure_gettext_initialized':

glibintl.h:37:42: warning: statement with no effect [-Wunused-value]

 #define bindtextdomain(Domain,Directory) (Domain)

                                          ^

ggettext.c:109:7: note: in expansion of macro 'bindtextdomain'

       bindtextdomain (GETTEXT_PACKAGE, GLIB_LOCALE_DIR);

       ^~~~~~~~~~~~~~

ar: `u' modifier ignored since `D' is the default (see `U')

+ make install prefix=/var/vcap/data/tmp/tmp.qVucM4AA3a

+ set -eu

+ tar xf gperf/gperf-3.1.tar.gz

+ cd gperf-3.1

+ ./configure --prefix=/var/vcap/packages/bpm-runc/gperf

+ make

+ make install

+ set -eu

+ export PATH=/var/vcap/packages/bpm-runc/gperf/bin:/var/vcap/bosh/bin:/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin

+ PATH=/var/vcap/packages/bpm-runc/gperf/bin:/var/vcap/bosh/bin:/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin

+ tar xf libseccomp/libseccomp-2.5.1.tar.gz

+ cd libseccomp-2.5.1

+ ./configure --prefix=/var/vcap/packages/bpm-runc/lib/libseccomp

+ make

ar: `u' modifier ignored since `D' is the default (see `U')

ar: `u' modifier ignored since `D' is the default (see `U')

+ make install

+ set -eu

+ export PATH=/var/vcap/data/tmp/tmp.qVucM4AA3a/bin:/var/vcap/bosh/bin:/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin

+ PATH=/var/vcap/data/tmp/tmp.qVucM4AA3a/bin:/var/vcap/bosh/bin:/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin

++ readlink -nf /var/vcap/packages/golang

+ export GOROOT=/var/vcap/data/packages/golang/3e624201b0bc67e918c0780991569e3b996ba53b

+ GOROOT=/var/vcap/data/packages/golang/3e624201b0bc67e918c0780991569e3b996ba53b

+ export PATH=/var/vcap/data/packages/golang/3e624201b0bc67e918c0780991569e3b996ba53b/bin:/var/vcap/data/tmp/tmp.qVucM4AA3a/bin:/var/vcap/bosh/bin:/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin

+ PATH=/var/vcap/data/packages/golang/3e624201b0bc67e918c0780991569e3b996ba53b/bin:/var/vcap/data/tmp/tmp.qVucM4AA3a/bin:/var/vcap/bosh/bin:/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin

+ export GOPATH=/var/vcap/packages/bpm-runc

+ GOPATH=/var/vcap/packages/bpm-runc

+ mkdir -p /var/vcap/packages/bpm-runc/src/github.com/opencontainers/runc

+ tar xf runc/runc-1.0.0-rc95.tar.xz -C /var/vcap/packages/bpm-runc/src/github.com/opencontainers/runc --strip-components 1

+ set -eu

+ pushd /var/vcap/packages/bpm-runc/src/github.com/opencontainers/runc

+ env GOCACHE=/var/vcap/packages/bpm-runc/cache PKG_CONFIG_PATH=/var/vcap/packages/bpm-runc/lib/libseccomp/lib/pkgconfig go build -buildmode=pie -ldflags '-X main.gitCommit=unknown -X main.version=1.0.0-rc95-bpm -r /var/vcap/packages/bpm/lib/libseccomp/lib' -tags 'seccomp apparmor' -o /var/vcap/packages/bpm-runc/bin/runc github.com/opencontainers/runc

**go: github.com/checkpoint-restore/go-criu/v5@v5.0.0: Get "https://proxy.golang.org/github.com/checkpoint-restore/go-criu/v5/@v/v5.0.0.mod": dial tcp: lookup proxy.golang.org on 10.186.16.35:53: no such host**



Task 51 Started  Tue May 25 00:04:40 UTC 2021

Task 51 Finished Tue May 25 00:22:06 UTC 2021

Task 51 Duration 00:17:26

Task 51 error


Updating deployment:

  Expected task '51' to succeed but state is 'error'


Exit code 1
```

**Comment by julweber** [comment]
+1

**Comment by julweber** [comment]
We did some additional research.

It seems that the problem is caused by the go-criu dependency in the runc package.

Since runc release v1.0.0-rc94 the go-criu package was updated to v5 5.0.0 (github.com/checkpoint-restore/go-criu/v5 v5.0.0)

We succesfully created and deployed a dev release of bpm using the runc release v1.0.0-rc93, which uses the go-criu v4.

I am not sure what causes the problem exactly and why the go-criu version triggers a download even if the vendor directory contains the go-criu module. Maybe this information helps in debugging the issue.

Kind regards,
Julian Weber

**Comment by pcf-user** [comment]
+1

**Comment by pcf-user** [comment]
Hi @ystros @rkoster  and rest of the BPM team -  Do we have any ETA on this issue?

**Comment by jpalermo** [comment]
Fixed in release 1.1.12.

Sorry for the delay, we didn't have this github repo wired up to notify us of issues. We'll get that fixed too so we don't miss these in the future.

**Comment by julweber** [comment]
@jpalermo @rkoster Thanks a lot guys. 👯
